dockstore-client/src/main/java/io/dockstore/client/cli/nested/ToolClient.java

```diff
@@ -714,6 +707,58 @@ public File downloadTargetEntry(String toolpath, ToolDescriptor.TypeEnum type, b
         }
     }
 
+    /**
+     * Returns the descriptors of the specified tool
+     * @param type Descriptor type CWL, WDL, NFL ...
+     * @param entryPath Workflow path
+     * @param versionID version we are getting descriptors for
+     */
+    @Override
+    public List<ToolFile> getAllToolDescriptors(String type, String entryPath, String versionID) {
+
+        final Ga4Ghv20Api ga4ghv20api = this.client.getGa4Ghv20Api();
+
+        // get all the tool files and filter out anything not a descriptor
+        try {
+            return ga4ghv20api.toolsIdVersionsVersionIdTypeFilesGet(type, entryPath, versionID).stream()
+                .filter(toolFile -> toolFile.getFileType().equals(ToolFile.FileTypeEnum.SECONDARY_DESCRIPTOR) || toolFile.getFileType().equals(ToolFile.FileTypeEnum.PRIMARY_DESCRIPTOR))
+                .collect(Collectors.toList());
+        } catch (NullPointerException ex) {
+            exceptionMessage(ex, "Unable to locate entry " + entryPath + ":" + versionID, Client.COMMAND_ERROR);
+        }
+
+        return null;
+    }
+
+    /**
+     * Returns the version ID for the provided entry path
+     * @param entryPath Tool path
+     */
+    @Override
+    public String getVersionID(String entryPath) {
+        final String[] parts = entryPath.split(":");
+
+        String versionID = parts.length > 1 ? parts[1] : null;
+
+        if (versionID == null) {
+            final DockstoreTool container = getDockstoreTool(parts[0]);
+            versionID = container.getDefaultVersion();
+        }
+
+        // last resort use 'latest' tag
+        return versionID == null ? "latest" : versionID;
+    }
+
+    /**
+     * Returns if the provided tool path is for a published entry
+     * @param entryPath Tool path
+     */
+    @Override
+    public boolean isEntryPublished(String entryPath) {
```

Comment:
I haven't addressed this in the current implementation yet. 

Because TRS endpoints only work for published entries, what is the desired way to handle this? I can make an API call for the remote descriptor file (full-text version), and locally calculate the SHA-1 checksum? ([Instead of using the GA4GHV20 call](https://github.com/dockstore/dockstore-cli/blob/7f6c85a4a186819af1abeaf2f795939c1fb693b4/dockstore-client/src/main/java/io/dockstore/client/cli/nested/BaseLanguageClient.java#L234))

Reply:
One other option, extend TRS endpoint to accept auth to access unpublished workflows (might be a big task that flows into a different ticket)